test/modes/ModeEstimation.jl

```diff
@@ -41,6 +42,31 @@ include(dir*"/test/test_utils/AllUtils.jl")
         @test all(isapprox.(m4.values.array - true_value, 0.0, atol=0.01))
     end
 
+    @testset "AD backends" begin
+        Random.seed!(222)
+        true_value = [0.0625, 1.75]
+        
+        Turing.setadbackend(:forwarddiff)
+        m1 = optimize(gdemo_default, MLE())
+        
+        Turing.setadbackend(:reversediff)
+        m2 = optimize(gdemo_default, MLE())
+
+        Turing.setadbackend(:tracker)
+        m3 = optimize(gdemo_default, MLE())
+
+        Turing.setadbackend(:zygote)
+        m4 = optimize(gdemo_default, MLE())
+
+        # Go back to normal forwarddiff for the rest of the tests
+        Turing.setadbackend(:forwarddiff)
+
+        @test all(isapprox.(m1.values.array - true_value, 0.0, atol=0.01))
+        @test all(isapprox.(m2.values.array - true_value, 0.0, atol=0.01))
+        @test all(isapprox.(m3.values.array - true_value, 0.0, atol=0.01))
+        @test all(isapprox.(m4.values.array - true_value, 0.0, atol=0.01))
+    end
+
```

Comment:
Maybe it would be better to move the mode tests to the AD block in runtests.jl, to more easily comment out tests of different AD backends?

(In general, IMO in a different PR we might want to restructure the tests similar to DistributionsAD or Bijectors such that we can test specific AD backends more easily using environment variables. This would also allow to split the tests in CI tests for each backend.)

Reply:
I'd prefer to keep it here because it falls under the optional dependency Optim -- there's really no great place to put these, so I think it's better to keep all of these "feature" tests under one banner.

Reply:
Oh maybe I didn't explain clearly what I mean. I just thought that instead of switching between AD backends here, you could move the `include(...)` line for the mode tests in runtests.jl inside of the block that cycles through the AD backends.

Reply:
Then you would not have to add any new tests at all, the mode tests would run with every AD backend automatically.

Reply:
Oh, yeah, that's way better. Done.